### PR TITLE
Make repo_prefix configurable

### PR DESCRIPTION
**Pull request by sohaibiftikhar**
Attempts fixing #38.

Allow repository_name to have a configurable prefix.

### CODE REVIEW

File: src/piptool.py [hunk 1]
```diff
@@ -62,18 +62,19 @@ def as_tuple(preq):
...
+    :param version: package version
     :param python_version: python major version
     :returns: repo name
     :rtype: str
 
     """
-    canonical = "pypi__{}__{}_{}".format(python_version, name, version)
-    return clean_name(canonical)
+    canonical = "__{}__{}_{}".format(python_version, name, version)
+    return f"{repo_prefix}{clean_name(canonical)}"
```
Comment (sohaibiftikhar) [inline src/piptool.py:77]: `repo_prefix` is deliberately left out of sanitisation cause it passes through as the user wants it. This was the only way I can get around stuff like python packages looking for `site-packages`.
If there is a better idea I am happy to be corrected.

Comment (sohaibiftikhar) [reply src/piptool.py:77]: TODO: Make this python2 compatible.

[the remark this answers is not in the document]
Comment (sohaibiftikhar) [reply src/piptool.py]: Done.

File: src/piptool.py [hunk 2]
```diff
@@ -140,6 +141,13 @@ def main():
         help=("The requirements.bzl file to export."),
         required=True,
     )
+    parser.add_argument(
+        "--repo-prefix",
+        action="store",
+        help=("The prefix to add to the reposuitory name for bazel."),
+        type=str,
+        required=True,
```
[the remark this answers is not in the document]
Comment (sohaibiftikhar) [reply src/piptool.py:149]: The default picked up through `defs.bzl`? Should I add the default in both places?

**Comment by ali5h** [comment]
would u rebase?

**Comment by sohaibiftikhar** [comment]
Sorry got busy with some other stuff. Done.

**Pull request merged**